src/ui/public/vislib/visualizations/point_series/heatmap_chart.js

```diff
@@ -150,7 +150,7 @@ export default function HeatmapChartFactory(Private) {
             val = colorsNumber - 1;
           } else {
             val = (d.y - min) / (max - min); /* get val from 0 - 1 */
-            val = Math.floor(val * (colorsNumber - 1));
+            val = Math.round(val * (colorsNumber - 1));
```

Comment:
Doing `(colorsNumber -1)` here before multiplying by the `val` is still causing issues. 

Given: colorNumbers=4  min=0  max=900  y=749
Buckets: 0=0-225  1=225-450  2=675  3=675-900
((749 - 0) / (900-0)) * (4-1) = 2.496 which rounds to 2